src/glview/cgal/CGALRenderUtils.cc

```diff
@@ -21,3 +50,23 @@ double calculateLineLineDistance(const Vector3d &l1b, const Vector3d &l1e, const
 	dist_lat=(v2.cross(n)).dot(l2b-l1b)/t;
 	return d;
 }
+
+double calculateSegSegDistance(const Vector3d &l1b, const Vector3d &l1e, const Vector3d &l2b, const Vector3d &l2e, double &dist_lat)
+{
+	double d;
+	Vector3d v1=l1e-l1b;
+	Vector3d v2=l2e-l2b;
+	Vector3d n=v1.cross(v2);
+	Vector3d res;
+	if(n.norm() < 1e-6) {
+		return calculateLinePointDistance(l1b,l1e,l2b,d);
+	}
+	if(linsystem(v1,n,v2,l2e-l1b,res,nullptr)) return NAN;
+	double d1=res[0], d2=res[2];
+	if(d1 < 0) d1=0;
+	if(d1 > 1) d1=1;
```

Comment:
I think you could just do `d1 = std::clamp(res[0], 0, 1);` to make things a tiny bit easier to read.